src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
-            }
+string JasmineGraphInstanceService::requestPerformanceStatistics(std::string isVMStatManager,
+                                                                 std::string isResourceAllocationRequested) {
+    int memoryUsage = collector.getMemoryUsageByProcess();
+    double cpuUsage = collector.getCpuUsage();
+    double loadAverage = collector.getLoadAverage();
+    std::string vmLevelStatistics = collector.collectVMStatistics(isVMStatManager, isResourceAllocationRequested);
+    auto executedTime = std::chrono::system_clock::now();
+    std::time_t reportTime = std::chrono::system_clock::to_time_t(executedTime);
+    std::string reportTimeString(std::ctime(&reportTime));
+    reportTimeString = Utils::trim_copy(reportTimeString, " \f\n\r\t\v");
+    std::string usageString =
+        reportTimeString + "," + to_string(memoryUsage) + "," + to_string(cpuUsage) + "," + to_string(loadAverage);
+    if (!vmLevelStatistics.empty()) {
+        usageString = usageString + "," + vmLevelStatistics;
+    }
+    return usageString;
+}
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+void JasmineGraphInstanceService::collectTrainedModels(
+    instanceservicesessionargs *sessionargs, std::string graphID,
+    std::map<std::string, JasmineGraphInstanceService::workerPartitions> graphPartitionedHosts, int totalPartitions) {
+    int total_threads = totalPartitions;
+    std::thread *workerThreads = new std::thread[total_threads];
+    int count = 0;
+    std::map<std::string, JasmineGraphInstanceService::workerPartitions>::iterator mapIterator;
+    for (mapIterator = graphPartitionedHosts.begin(); mapIterator != graphPartitionedHosts.end(); mapIterator++) {
+        string hostName = mapIterator->first;
+        JasmineGraphInstanceService::workerPartitions workerPartitions = mapIterator->second;
+        std::vector<std::string>::iterator it;
+        for (it = workerPartitions.partitionID.begin(); it != workerPartitions.partitionID.end(); it++) {
+            workerThreads[count] =
+                std::thread(&JasmineGraphInstanceService::collectTrainedModelThreadFunction, sessionargs, hostName,
+                            workerPartitions.port, workerPartitions.dataPort, graphID, *it);
+            count++;
+        }
+    }
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphVertexCount = (data);
-            graphVertexCount = Utils::trim_copy(graphVertexCount, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID:" + graphID + " Vertex Count: " + graphVertexCount, "info");
+    for (int threadCount = 0; threadCount < count; threadCount++) {
+        workerThreads[threadCount].join();
+        std::cout << "Thread [C]: " << threadCount << " joined" << std::endl;
```

Comment:
I do not see a point of having this `cout` statement here.

Reply:
Removed in the next PR

Reply:
Ok